### DISCUSSION TITLE
Feature: make Umbraco backoffice scalable horizontally

### DISCUSSION THREAD

**Discussion by MarGraz** [umbraco/Umbraco-CMS#14866] [Features and ideas]
Hi,

do you know if in the near future the Umbraco team will make Umbraco backoffice scalable horizontally?

[According to the documentation](https://docs.umbraco.com/umbraco-cms/fundamentals/setup/server-setup/load-balancing#how-umbraco-load-balancing-works), right now only the frontend is scalable, but we need to scale the backoffice horizontally, but the problem is that all the DB calls in the Umbraco core are syncronous, and there's a problem of concurrency. 

We have a small project that needs to scale, and we don't want to create several frontend instances (they will costs a lot), but we prefer to scale the backoffice when needed.

Does anybody know if in the future they will make the backoffice scalable?

Thank you

I also asked [this question here](https://our.umbraco.com/forum/using-umbraco-and-getting-started//112738-feature-make-umbraco-backoffice-scalable-horizontally).

**Comment by bergmania** [comment]
Hi @MarGraz.

Currently this is not a main focus. We focus mainly on the new backoffice client at the moment..

We are aware this is a limitation of Umbraco and dream of to make it officially supported in the future

**Reply by MarGraz** [reply]
@bergmania thank you for your reply, I hope that you will add this topic in your agenda soon, because on small projects we can't use Umbraco, due to the high costs to have several frontend apps.

**Reply by miguelcrpinto** [reply]
@bergmania considering Umbraco's goal to grow in the market segment, I believe this is something that you should already be thinking about and prioritizing. 

I'm currently maintaining an Umbraco 8 multi site (dozens of sites and one of them has a few thousands of nodes) project for a big corporate and we are struggling when we have multiple editors trying to create/save/publish/move/copy content. We have already upscaled the infrastructure (running in a container on ECS on an EC2 with 8 vCPUs and 32gb of RAM!) and we still cannot cope with the number of editors.

I know that newer versions of Umbraco have better performance (unfortunately we cannot update at the moment) but I believe that even newer versions of Umbraco would not be able to handle so many simultaneous back office users. 

I think that there are two main issues:
- **Slow publishing of pages with block list editors containing a large number of blocks** - I've noticed that the more block list editors you have on a page and the larger the amount of blocks you have on each, the longer it takes publish a page. The API call to get the content by id (/umbraco/backoffice/UmbracoApi/Content/GetById?id={id}) is also slower in theses cases. I did not test the save, move or copy of such nodes but I think that it also takes longer when you have lots of content blocks.
- **Content tree write locks on content save/publish/move/copy actions** - I do understand why you need to create write locks on save/publish/move/copy but this combined with the previous issue (long publishing times) leads to a lock of lock timeout exceptions. I believe this is one of the biggest bottle neck in terms of performance and also prevents the back office from horizontally scaling.

I also had these same performance issues on another project hosted in Umbraco Heartcore but in this case with a lot less content and a lot less editors simultaneously active in the backoffice.

Please consider prioritizing this feature.

**Reply by ronaldbarendse** [reply]
Hi Miguel,

Please note that you can still serve frontend requests with the single backoffice instance, but you'd have to create a separate frontend instance to setup auto-scaling (to avoid also scaling the backoffice instance). See the last Q&A at the bottom of this documentation page: https://docs.umbraco.com/umbraco-cms/fundamentals/setup/server-setup/load-balancing. So you could start load-balancing with just 2 instances, a backoffice and frontend one, that both process frontend requests.

Newer versions the CMS are generally faster and the jump from .NET Framework to .NET Core was quite significant, so I'd recommend trying to upgrade to the latest if possible (especially with v13 LTS that's soon to be released).

Regarding the loading performance in the backoffice, could that be related to https://github.com/umbraco/Umbraco-CMS/issues/14936?

**Reply by miguelcrpinto** [reply]
@ronaldbarendse I'm aware that we can auto-scale the frontend and I've used that setup in other projects. The project in question does not have any frontend website being rendered by Umbraco, we have a custom setup kind of like what you do in Heartcore but with different infrastructure (we use MongoDB and Realm). Our struggle is really the backoffice.

Issue [14936](https://github.com/umbraco/Umbraco-CMS/issues/14936) certainly does affects us and we are trying port that fix back to V8 but even without that issue, if you have lots of editors performing save/publish/move/copy actions, you'll end up seeing lock timeout exceptions (caused by the content tree write locks).

Upgrading to newer version is not an option and even then, with number of editors we have, I suspect we would have lock issues.

I know this is not an easy problem to solve and that not all (or even most) of the umbraco projects suffer from but I believe it's something that should be prioritized if you want to have bigger sites/companies using Umbraco. And like I said above, this also affects your SaaS solution (Heartcore) where customers just expect the CMS to perform independent of the number of simultaneous editors.

**Comment by AndyButland** [comment]
Will close this one now as load balancing the backoffice is a feature of Umbraco 17: https://docs.umbraco.com/umbraco-cms/17.latest/fundamentals/setup/server-setup/load-balancing/load-balancing-backoffice
